### PR TITLE
{bio}[foss/2018b] STAR-Fusion v1.5.0 w/ Perl v5.28.0

### PR DESCRIPTION
**Pull request by jdemeul**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/s/STAR-Fusion/STAR-Fusion-1.5.0-foss-2018b-Perl-5.28.0.eb [hunk 1]
```diff
@@ -15,33 +16,59 @@ description = """STAR-Fusion uses the STAR aligner to identify candidate fusion
  to map junction reads and spanning reads to a reference annotation set."""
 
 toolchain = {'name': 'foss', 'version': '2018b'}
-
-source_urls = ['https://github.com/STAR-Fusion/STAR-Fusion/releases/download/%(name)s-v%(version)s/']
+source_urls = ['https://github.com/%(name)s/%(name)s/releases/download/%(name)s-v%(version)s/']
```
Comment (smoors) [inline easybuild/easyconfigs/s/STAR-Fusion/STAR-Fusion-1.5.0-foss-2018b-Perl-5.28.0.eb]: please add empty line above this one.

File: easybuild/easyconfigs/s/STAR-Fusion/STAR-Fusion-1.5.0-foss-2018b-Perl-5.28.0.eb [hunk 2]
```diff
@@ -15,33 +16,59 @@ description = """STAR-Fusion uses the STAR aligner to identify candidate fusion
...
 sources = ['%(name)s-v%(version)s.FULL.tar.gz']
 checksums = ['229cbf7980496d72de0bfab5bde7148ff3111a99b72c6169d1a04120a52508b4']
 
 dependencies = [
     ('Perl', '5.28.0'),
-    ('STAR', '2.6.1c'),
+    ('DB', '18.1.25'),
+    ('STAR', '2.7.0d', '', ('foss', '2018b')),
```
Comment (smoors) [inline easybuild/easyconfigs/s/STAR-Fusion/STAR-Fusion-1.5.0-foss-2018b-Perl-5.28.0.eb]: please replace with: `('STAR', '2.6.1c')`.
this is the single dependency version rule: version `2.6.1c` is already used as a dependency for the `2018b` toolchains.

File: easybuild/easyconfigs/s/STAR-Fusion/STAR-Fusion-1.5.0-foss-2018b-Perl-5.28.0.eb [hunk 3]
```diff
@@ -15,33 +16,59 @@ description = """STAR-Fusion uses the STAR aligner to identify candidate fusion
...
+        'checksums': ['cc1062e39a1fd6ff8bd9df84a102e8572468cea29d43190672a10f9c39973e5c'],
+    }),
+    ('PerlIO::gzip', '0.20', {
+        'source_tmpl': 'PerlIO-gzip-0.20.tar.gz',
+        'source_urls': ['https://cpan.metacpan.org/authors/id/N/NW/NWCLARK/'],
+        'checksums': ['4848679a3f201e3f3b0c5f6f9526e602af52923ffa471a2a3657db786bd3bdc5'],
+    }),
 ]
 
+
```
Comment (smoors) [inline easybuild/easyconfigs/s/STAR-Fusion/STAR-Fusion-1.5.0-foss-2018b-Perl-5.28.0.eb]: please remove empty line

File: easybuild/easyconfigs/s/STAR-Fusion/STAR-Fusion-1.5.0-foss-2018b-Perl-5.28.0.eb [hunk 4]
```diff
@@ -16,14 +16,15 @@ description = """STAR-Fusion uses the STAR aligner to identify candidate fusion
...
 source_urls = ['https://github.com/%(name)s/%(name)s/releases/download/%(name)s-v%(version)s/']
 sources = ['%(name)s-v%(version)s.FULL.tar.gz']
 checksums = ['229cbf7980496d72de0bfab5bde7148ff3111a99b72c6169d1a04120a52508b4']
 
 dependencies = [
     ('Perl', '5.28.0'),
     ('DB', '18.1.25'),
-    ('STAR', '2.7.0d', '', ('foss', '2018b')),
+    ('STAR', '2.6.1c', '', ('foss', '2018b')),
```
Comment (smoors) [inline easybuild/easyconfigs/s/STAR-Fusion/STAR-Fusion-1.5.0-foss-2018b-Perl-5.28.0.eb]: there is no need to add the toolchain here, so please replace with `('STAR', '2.6.1c')`

File: easybuild/easyconfigs/s/STAR-Fusion/STAR-Fusion-1.5.0-foss-2018b-Perl-5.28.0.eb [hunk 5]
```diff
@@ -30,16 +32,41 @@ exts_defaultclass = 'PerlModule'
 exts_filter = ("perldoc -lm %(ext_name)s ", "")
 
 exts_list = [
-    ('Set::IntervalTree', '0.10', {
-        'source_tmpl': 'Set-IntervalTree-0.10.tar.gz',
-        'source_urls': ['https://cpan.metacpan.org/authors/id/B/BE/BENBOOTH/'],
-        'checksums': ['e3bd9ccf0d074b5f879eef1ed88254983697bf83d02744fce62150ee46553ebc'],
+    ('ExtUtils::CppGuess', '0.12', {
```
Comment (smoors) [inline easybuild/easyconfigs/s/STAR-Fusion/STAR-Fusion-1.5.0-foss-2018b-Perl-5.28.0.eb]: @jdemeul out of curiosity: why we you need 'ExtUtils::CppGuess' here?

**Comment by smoors** [comment]
@jdemeul out of curiosity: why we need 'ExtUtils::CppGuess' here?

File: easybuild/easyconfigs/s/STAR-Fusion/STAR-Fusion-1.5.0-foss-2018b-Perl-5.28.0.eb (hunk 5 above)
Comment (jdemeul) [reply easybuild/easyconfigs/s/STAR-Fusion/STAR-Fusion-1.5.0-foss-2018b-Perl-5.28.0.eb]: @smoors Set::IntervalTree seems to require this.

When building without it:
`== installing extension Set::IntervalTree 0.12 (1/6)...
== FAILED: Installation ended unsuccessfully (build directory: /tmp/STARFusion/1.5.0/foss-2018b-Perl-5.28.0): build failed (first 300 chars): cmd " perl Makefile.PL PREFIX=/srv/sw/eb/software/STAR-Fusion/1.5.0-foss-2018b-Perl-5.28.0 " exited with exit code 2 and output:
Can't locate ExtUtils/CppGuess.pm in @INC (you may need to install the ExtUtils::CppGuess module) (@INC contains: /srv/sw/eb/software/STAR-Fusion/1.5.0-foss-2018b-Perl-5.2`

Happy to remove again if there's a better way to fix this?

**Comment by smoors** [comment]
Test report by @smoors
**FAILED**
Build succeeded for 0 out of 2 (1 easyconfigs in this PR)
nic413 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/5aea5da55f5be09500b799102b71089f for a full test report.

**Comment by smoors** [comment]
Test report by @smoors
**FAILED**
Build succeeded for 0 out of 2 (1 easyconfigs in this PR)
nic26 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/270d5cf8d077acd6283d3745702ed31a for a full test report.

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in this PR)
nic26 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/d49e1637dfd98ff1ccf57f3b3616486b for a full test report.

**Comment by smoors** [comment]
Going in, thanks @jdemeul!

**Pull request merged**